src/query/sql/src/planner/semantic/type_check.rs

```diff
@@ -650,7 +650,7 @@ impl<'a> TypeChecker<'a> {
                 params,
                 window,
                 lambda,
-            } => {
+            } if window.is_none() => {
```

Comment:
It is better to determine the type of the function by its name, and then determine whether the `window` should be `None` or `Some`. Separately handle will result in the window function not being able to find similar function names if the function name is not correct. We have code logic for finding similar function names between lines 662 and 695.
https://github.com/datafuselabs/databend/pull/12925/files#diff-8e24042e5eec79cd29dda9776333d9ecad8cba883a80e4590c31f594ea745795R662-R695